### PR TITLE
convert rxjava2 unit tests to java

### PR DESCRIPTION
**Pull request by kaibocai7**
convert rxjava2 unit tests to java

### CODE REVIEW

**Comment by kaibocai7** [comment]
Any idea how to resolve the `Gateway Error` in CI, seems they are not related to the change I made. Is there a way to trigger reruning the CI?

File: instrumentation/rxjava/rxjava-2.0/library/src/test/java/RxJava2SubscriptionTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class RxJava2SubscriptionTest extends AbstractRxJava2SubscriptionTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = LibraryInstrumentationExtension.create();
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  static TracingAssembly tracingAssembly = TracingAssembly.create();
```
[the remark this answers is not in the document]
Comment (kaibocai7) [reply instrumentation/rxjava/rxjava-2.0/library/src/test/java/RxJava2SubscriptionTest.java:22]: The method annotated by @BeforeAll is suggested be static, so this field I made it static.

File: instrumentation/rxjava/rxjava-2.0/testing/src/main/java/io/opentelemetry/instrumentation/rxjava/v2_0/AbstractRxJava2Test.java [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+    assertThat(result).isEqualTo(4);
+    testing()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> span.hasName(PARENT).hasKind(SpanKind.INTERNAL).hasNoParent(),
+                    span ->
+                        span.hasName(ADD_ONE)
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasParent(trace.getSpan(0)),
```
[the remark this answers is not in the document]
Comment (kaibocai7) [reply instrumentation/rxjava/rxjava-2.0/testing/src/main/java/io/opentelemetry/instrumentation/rxjava/v2_0/AbstractRxJava2Test.java:121]: Thank you for your review and comments. I just think that explicitly writing out the test cases in each section will make it easier to understand and quickly identify what is being tested.

[the remark this answers is not in the document]
Comment (breedx-splk) [reply instrumentation/rxjava/rxjava-2.0/testing/src/main/java/io/opentelemetry/instrumentation/rxjava/v2_0/AbstractRxJava2Test.java:121]: Fair enough, I also usually favor duplication, but this was pretty heavy. I think it's fine tho. 👍🏻

**Comment by trask** [comment]
thx @kaibocai7!

**Pull request merged**